### PR TITLE
Add pcaspy

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This adds a recipe for `pcaspy`, modified from the original at
https://github.com/paulscherrerinstitute/pcaspy/tree/2e57546c094bf002599dfd889c02041bb7561b7a/conda-recipe

cc @simongregorebner, @tacaswell (possibly interested maintainers)

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@conda-forge/help-python-c, the [windows build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=39798) presents lot of warnings like this:

```
WARNING (pcaspy,Lib\site-packages\pcaspy\cas.dll): $RPATH\ca.dll not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
```

`ca.dll` is provided by `epics-base`, is there something I can add to the recipe to silence/prevent this warnings?

[I am extremely windows naïve, apologies if this is basic stuff]

**Comment by tacaswell** [comment]
I am less interest in being on this one, when I need an IOC from python I reach for https://github.com/caproto/caproto

File: recipes/pcaspy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+package:
+  name: pcaspy
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/p/pcaspy/pcaspy-{{ version }}.tar.gz
+  sha256: 823e5b3c25dcb73795730aee4f656e21fb4f590f866fcf6adb2d56cc6b5ced3d
+  patches:
+    - readline.patch  # [win]
```
Comment (marcelotrevisani) [inline recipes/pcaspy/meta.yaml]: Could you please write some explanation why this patch is needed?

Comment (duncanmmacleod) [reply recipes/pcaspy/meta.yaml]: @marcelotrevisani, see 42bc982

**Pull request merged**